packages/components/src/VirtualizedList/utils/resizable.js

```diff
@@ -0,0 +1,267 @@
+ * and update the corresponding column with computed width.
+ * @param {number} deltaX
+ * @param {integer} index
+ */
+const shrinkLeftColumn = (deltaX, index) => columnsWidths => {
+	const shrinkIndexLeft = getShrinkIndexLeft(index, columnsWidths);
+	if (shrinkIndexLeft >= 0) {
+		flow([
+			calculateWidth(subtractWidth(Math.abs(deltaX))),
+			setResized(true),
+			setColumn(columnsWidths, shrinkIndexLeft),
+		])(columnsWidths[shrinkIndexLeft]);
+	}
+	return columnsWidths;
+};
+
+/**
+ * Get the nearest column index to enlarge on the right side of the given index,
+ * and update the corresponding column with computed width.
+ * @param {number} deltaX
+ * @param {integer} index
+ */
+const enlargeRightColumn = (deltaX, index) => columnsWidths => {
+	const enlargeIndexRight = getEnlargeIndexRight(index, columnsWidths);
+	if (enlargeIndexRight >= 0) {
+		flow([
+			calculateWidth(addWidth(Math.abs(deltaX))),
+			setResized(true),
+			setColumn(columnsWidths, enlargeIndexRight),
+		])(columnsWidths[enlargeIndexRight]);
+	}
+	return columnsWidths;
+};
+
+/**
+ * Get the nearest column index to shrink on the right side of the given index,
+ * and update the corresponding column with computed width.
+ * @param {number} deltaX
+ * @param {integer} index
+ */
+const shrinkRightColumn = (deltaX, index) => columnsWidths => {
+	const shrinkIndexRight = getShrinkIndexRight(index, columnsWidths);
+	if (shrinkIndexRight >= 0) {
+		flow([
+			calculateWidth(subtractWidth(deltaX)),
+			setResized(true),
+			setColumn(columnsWidths, shrinkIndexRight),
+		])(columnsWidths[shrinkIndexRight]);
+	}
+	return columnsWidths;
+};
+
+/**
+ * Enlarge the column index.
+ * @param {number} deltaX
+ * @param {integer} index
+ */
+const enlargeCurrentColumn = (deltaX, index) => columnsWidths => {
+	if (index >= 0) {
+		flow([calculateWidth(addWidth(deltaX)), setResized(true), setColumn(columnsWidths, index)])(
+			columnsWidths[index],
+		);
+	}
+	return columnsWidths;
+};
+
+/**
+ * Flow of operations to handle dragging column to the right.
+ * @param {number} deltaX
+ * @param {integer} index
+ */
+const resizeRight = (deltaX, index) => columnsWidths => {
+	if (deltaX >= 0) {
+		flow([enlargeCurrentColumn(deltaX, index), shrinkRightColumn(deltaX, index)])(columnsWidths);
+	}
+	return columnsWidths;
+};
+
+/**
+ * Flow of operations to handle dragging column to the right.
+ * @param {number} deltaX
+ * @param {integer} index
+ */
+const resizeLeft = (deltaX, index) => columnsWidths => {
+	if (deltaX < 0) {
+		flow([shrinkLeftColumn(deltaX, index), enlargeRightColumn(deltaX, index)])(columnsWidths);
+	}
+	return columnsWidths;
+};
+
+/**
+ * Clone the incoming collection, mutate it, and return new widths value.
+ * @param {number} deltaX
+ * @param {array} columnsWidths
+ * @param {integer} currentIndex
+ */
+export const resizeColumns = (deltaX, columnsWidths, currentIndex) =>
+	flow([resetResized, resizeRight(deltaX, currentIndex), resizeLeft(deltaX, currentIndex)])(
```

Comment:
There are missing steps in your algorithm. That introduce issues : note the width that is increased continually.
![infinite-resize](https://user-images.githubusercontent.com/10761073/60256256-7a5cce00-98d1-11e9-99d0-3fc294acd156.gif)

What should be done :
* calculate the size of each col depending on the deltaX. This should take all the widths in a global algorithm. So if you want to enlarge a col, but there is not col on the right that can be shrinked, nothing should be changed for example.
* apply the new width

Reply:
already talked with you about that...
coming just after this pr, already in progress